control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -19,39 +19,305 @@ package broker
-	)
+func incrementVolumeGeneration(generation uint64) uint64 {
+	return (generation + 1) % (math.MaxUint64 - 1)
+}
+
+func (r *Reconciler) createTopic(broker *eventing.Broker) (string, error) {
+	topic := topic(broker)
+	topicDetail := r.topicDetailFromBrokerConfig(broker)
+
+	r.KafkaClusterAdminLock.Lock()
+	defer r.KafkaClusterAdminLock.Unlock()
+
+	return topic, r.KafkaClusterAdmin.CreateTopic(topic, topicDetail, true)
+}
+
+func (r *Reconciler) deleteTopic(broker *eventing.Broker) (string, error) {
+
+	r.KafkaClusterAdminLock.RLock()
+	defer r.KafkaClusterAdminLock.RUnlock()
+
+	topic := topic(broker)
+	return topic, r.KafkaClusterAdmin.DeleteTopic(topic)
+}
+
+func topic(broker *eventing.Broker) string {
+	return fmt.Sprintf("%s%s-%s", TopicPrefix, broker.Namespace, broker.Name)
+}
+
+func (r *Reconciler) topicDetailFromBrokerConfig(broker *eventing.Broker) *sarama.TopicDetail {
+	// TODO use broker configurations - see https://github.com/knative-sandbox/eventing-kafka-broker/issues/9
+
+	r.KafkaDefaultTopicDetailsLock.RLock()
+	defer r.KafkaDefaultTopicDetailsLock.RUnlock()
+
+	// copy the default topic details
+	topicDetail := r.KafkaDefaultTopicDetails
+	return &topicDetail
+}
+
+func (r *Reconciler) getBrokerConfig(topic string, broker *eventing.Broker) (*coreconfig.Broker, error) {
+	brokerConfig := &coreconfig.Broker{
+		Id:    string(broker.UID),
+		Topic: topic,
+	}
+
+	if broker.Spec.Delivery == nil || broker.Spec.Delivery.DeadLetterSink == nil {
+		return brokerConfig, nil
+	}
+
+	deadLetterSinkURL, err := r.Resolver.URIFromDestinationV1(*broker.Spec.Delivery.DeadLetterSink, broker)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve broker.Spec.Deliver.DeadLetterSink: %w", err)
+	}
+
+	brokerConfig.DeadLetterSink = deadLetterSinkURL.String()
+
+	return brokerConfig, nil
+}
+
+// SetBootstrapServers change kafka bootstrap brokers addresses.
+// servers: a comma separated list of brokers to connect to.
+func (r *Reconciler) SetBootstrapServers(servers string) error {
+	addrs := strings.Split(servers, ",")
+
+	kafkaClusterAdmin, err := NewClusterAdmin(addrs, sarama.NewConfig())
+	if err != nil {
+		return fmt.Errorf("failed to create kafka cluster admin: %w", err)
+	}
+
+	r.KafkaClusterAdminLock.Lock()
+	oldKafkaClusterAdmin := r.KafkaClusterAdmin
+
+	// first unlock and then close the old one.
+	defer func() {
+		if oldKafkaClusterAdmin != nil {
+			_ = oldKafkaClusterAdmin.Close()
+		}
+	}()
+	defer r.KafkaClusterAdminLock.Unlock()
+
+	r.KafkaClusterAdmin = kafkaClusterAdmin
+	return nil
+}
+
+func (r *Reconciler) SetDefaultTopicDetails(topicDetail sarama.TopicDetail) {
+	r.KafkaDefaultTopicDetailsLock.Lock()
+	defer r.KafkaDefaultTopicDetailsLock.Unlock()
+
+	r.KafkaDefaultTopicDetails = topicDetail
+}
+
+func findBroker(brokersTriggers *coreconfig.Brokers, broker *eventing.Broker) int {
+	// Find broker in brokersTriggers.
+	brokerIndex := noBroker
+	for i, b := range brokersTriggers.Broker {
```

Comment:
Can you index the brokers with the UID directly more than storing them as array? If that requires a change in the proto schema, then do it in another PR, opening an issue for that